### DISCUSSION TITLE
How to push only the required files and exclude unnecessary files (e.g., package.json and package-lock.json)?

### DISCUSSION THREAD

**Discussion by PiyushKalyanpy** [PiyushKalyanpy/GyanaGuru#466] [Q&A] [answered]
As most of the contributors have been pushing unnecessary files, like package files, it is crucial that we adopt a more focused approach. To ensure efficient collaboration, each contributor should push only the files directly addressing the assigned issues.

Please specify the steps and code which can solve this problem.

**Comment by Nikhil1017** [comment, marked as answer]
1) The simplest way to is to use the Discard button in Github extension for VS code

  ![image](https://github.com/PiyushKalyanpy/GyanaGuru/assets/115337705/41a80f0a-0a97-44a7-83a3-aea66b2e8be9)

This will undo all the changes for a specific file so that it will not be included in the PR

2) Another way is to use the CLI  command 
 ```
  git restore filename
 ```
 For Example  
 ```
  git restore package.json 
 ```
This will undo all changes for package.json.

3) If for some reason, you want to keep the package.json file after pushing, you can follow this procedure. First before pushing use the command 
 ```
  git stash push package.json
 ```
 This will undo the changes, now push the PR.
 Now after pushing the PR use the command 
 ```
 git stash pop 
 ```
This will bring back the changes in your local repository.
